pkg/cli/admin/nodeimage/create.go

```diff
@@ -0,0 +1,578 @@
+				},
+			},
+			{
+				APIGroups: []string{
+					"",
+				},
+				Resources: []string{
+					"configmaps",
+					"nodes",
+					"secrets",
+				},
+				Verbs: []string{
+					"get",
+					"list",
+				},
+			},
+		},
+	}
+	cr, err := o.Client.RbacV1().ClusterRoles().Create(ctx, nodeJoinerRole, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot create role: %w", err)
+	}
+	o.nodeJoinerRole = cr
+
+	nodeJoinerRoleBinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "node-joiner-",
+			Annotations: map[string]string{
+				"oc.openshift.io/command": "oc adm node-image create",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "v1",
+					Kind:       "Namespace",
+					Name:       o.nodeJoinerNamespace.GetName(),
+					UID:        o.nodeJoinerNamespace.GetUID(),
+				},
+			},
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      o.nodeJoinerServiceAccount.GetName(),
+				Namespace: o.nodeJoinerNamespace.GetName(),
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     o.nodeJoinerRole.GetName(),
+		},
+	}
+	_, err = o.Client.RbacV1().ClusterRoleBindings().Create(ctx, nodeJoinerRoleBinding, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot create role binding: %w", err)
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) createInputConfigMap(ctx context.Context) error {
+	data, err := fs.ReadFile(o.FSys, nodeJoinerConfigurationFile)
+	if err != nil {
+		return err
+	}
+
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nodes-config",
+			Namespace: o.nodeJoinerNamespace.GetName(),
+		},
+		Data: map[string]string{
+			nodeJoinerConfigurationFile: string(data),
+		},
+	}
+
+	_, err = o.Client.CoreV1().ConfigMaps(o.nodeJoinerNamespace.GetName()).Create(ctx, cm, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot create configmap: %w", err)
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) createPod(ctx context.Context) error {
+	nodeJoinerPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "node-joiner-",
+			Labels: map[string]string{
+				"app": "node-joiner",
+			},
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy:      corev1.RestartPolicyNever,
+			ServiceAccountName: o.nodeJoinerServiceAccount.GetName(),
+			SecurityContext: &corev1.PodSecurityContext{
+				SeccompProfile: &corev1.SeccompProfile{
+					Type: corev1.SeccompProfileTypeRuntimeDefault,
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "nodes-config",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "nodes-config",
+							},
+						},
+					},
+				},
+				{
+					Name: "assets",
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
```

Comment:
Maybe adding some constraints to limit the maximum allocated storage, especially for the limited environments.

Reply:
Thanks for spotting it, we also defined it in the original [script](https://github.com/openshift/installer/blob/a8b4b1cc6baef97f3dfe16cd4f2eb35fe0374f80/docs/user/agent/add-node/node-joiner.sh#L120) but somehow missed :)